mlir/lib/Conversion/FluxToQuartz/FluxToQuartz.cpp

```diff
@@ -0,0 +1,414 @@
+ * alongside the measurement bit. MLIR's conversion infrastructure automatically
+ * routes subsequent uses of the Flux output qubit to this Quartz reference.
+ *
+ * Register metadata (name, size, index) for output recording is preserved
+ * during conversion.
+ *
+ * Example transformation:
+ * ```mlir
+ * %q_out, %c = flux.measure("c", 2, 0) %q_in : !flux.qubit
+ * // becomes:
+ * %c = quartz.measure("c", 2, 0) %q : !quartz.qubit -> i1
+ * // %q_out uses are replaced with %q (the adaptor-converted input)
+ * ```
+ */
+struct ConvertFluxMeasureOp final : OpConversionPattern<flux::MeasureOp> {
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(flux::MeasureOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter& rewriter) const override {
+    // OpAdaptor provides the already type-converted input qubit
+    const auto& quartzQubit = adaptor.getQubitIn();
+
+    // Create quartz.measure (in-place operation, returns only bit)
+    // Preserve register metadata for output recording
+    auto quartzOp = rewriter.create<quartz::MeasureOp>(
+        op.getLoc(), quartzQubit, op.getRegisterNameAttr(),
+        op.getRegisterSizeAttr(), op.getRegisterIndexAttr());
+
+    auto measureBit = quartzOp.getResult();
+
+    // Replace both results: qubit output → same quartz reference, bit → new bit
+    rewriter.replaceOp(op, {quartzQubit, measureBit});
+
+    return success();
+  }
+};
+
+/**
+ * @brief Converts flux.reset to quartz.reset
+ *
+ * @details
+ * Reset operations force a qubit to the |0⟩ state:
+ * - Flux (value semantics): Consumes input qubit, returns reset output qubit
+ * - Quartz (reference semantics): Resets qubit in-place, no result value
+ *
+ * The OpAdaptor provides the input qubit already converted to !quartz.qubit.
+ * Since Quartz's reset is in-place, we return the same qubit reference.
+ * MLIR's conversion infrastructure automatically routes subsequent uses of
+ * the Flux output qubit to this Quartz reference.
+ *
+ * Example transformation:
+ * ```mlir
+ * %q_out = flux.reset %q_in : !flux.qubit -> !flux.qubit
+ * // becomes:
+ * quartz.reset %q : !quartz.qubit
+ * // %q_out uses are replaced with %q (the adaptor-converted input)
+ * ```
+ */
+struct ConvertFluxResetOp final : OpConversionPattern<flux::ResetOp> {
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(flux::ResetOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter& rewriter) const override {
+    // OpAdaptor provides the already type-converted input qubit
+    const auto& quartzQubit = adaptor.getQubitIn();
+
+    // Create quartz.reset (in-place operation, no result)
+    rewriter.create<quartz::ResetOp>(op.getLoc(), quartzQubit);
+
+    // Replace the output qubit with the same quartz reference
+    rewriter.replaceOp(op, quartzQubit);
+
+    return success();
+  }
+};
+
+// Temporary implementation of XOp conversion
```

Comment:
As noted in the other conversion I think these are rather here to stay, so they would probably deserve proper docstrings.
One could try and define them as patterns in the tablegen files, which might reduce the amount of code that needs to be written.